contracts/rbn-staking/PenaltyDistributor.vy

```diff
@@ -418,11 +433,31 @@ def donate(_amount: uint256) -> bool:
 
     ERC20(self.token).transferFrom(msg.sender, self, _amount)
 
+    penalty_rebate: uint256 = self.penalty_rebate_of[tx.origin]
```

Comment:
I'm not too familiar with vyper contracts, and this might not be relevant, but is it possible to add reentrancy guards for this? Just want to make sure no issues here in light of recent issues with donate functions